### ISSUE TITLE
[20.03,master] epkowa: driver tarballs hash mismatch

### ISSUE THREAD

**Issue by pacien** [NixOS/nixpkgs#93812]
**Describe the bug**

Some tarballs downloaded from Epson's website don't match the pinned hashes anymore.

**To Reproduce**

Try to build `pkgs.epkowa` on master or release-20.03:

```
❯ NIXPKGS_ALLOW_UNFREE=1 nix-build -A epkowa
these derivations will be built:
  /nix/store/4rvdspgkw2kznzqyiq5y1al2bwbwj2p9-iscan-gt-x770-bundle-2.30.4.x64.rpm.tar.gz.drv
  /nix/store/02lw4k8mpi9wrjvcksjbqn6q98b5ab0s-iscan-gt-x770-bundle-2.30.4.drv
  /nix/store/qf1sv0zy8prp39v74wjqgcrb41570mmi-iscan-gt-f720-bundle-2.30.4.x64.rpm.tar.gz.drv
  /nix/store/icycj83jf7lin6gqb6q9h4z8nrslrdj7-iscan-gt-f720-bundle-2.30.4.drv
  /nix/store/gfx46padnajbdg6b21az60wiic3c87c3-iscan-gt-s650-bundle-2.30.4.x64.rpm.tar.gz.drv
  /nix/store/k4g0rzh0rn0km6gr1ylg5cd28r06lf9s-iscan-gt-s650-bundle.drv
  /nix/store/sgfqy0l7zxaqn59zii5v5ghb79p4m1ns-iscan-perfection-v330-bundle-2.30.4.x64.rpm.tar.gz.drv
  /nix/store/l32c3r0i2c8w5ld25kp32gkl2g2h54vz-iscan-v330-bundle.drv
  /nix/store/8f9gb8qlz0wx0pc2n18pafmnmn717lvf-iscan-bundle-2.30.4.x64.rpm.tar.gz.drv
  /nix/store/n23vabjdm2nhkmh5y4i5w7qmzwk7z9ij-iscan-nt-bundle-2.30.4.drv
  /nix/store/3cv0902px1zhgbpqff24zgs4jw3vpdgj-iscan-gt-s80-bundle-2.30.4.x64.rpm.tar.gz.drv
  /nix/store/nx68d6n4w4lmn50kdibf50a4391j9b7d-iscan-gt-s80-bundle-2.30.4.drv
  /nix/store/12b0kblg7rsvs17z6q8akw8pnwky3rsz-esci-firmware-dir.drv
  /nix/store/7yff041q51pva32lbqix3hjb7c6npcbl-iscan-data-1.39.1-2.drv
  /nix/store/k3j68jk2ssyg82q44kp5gmnxmk0z4136-iscan-2.30.4-2.drv
building '/nix/store/7yff041q51pva32lbqix3hjb7c6npcbl-iscan-data-1.39.1-2.drv'...
building '/nix/store/8f9gb8qlz0wx0pc2n18pafmnmn717lvf-iscan-bundle-2.30.4.x64.rpm.tar.gz.drv'...
building '/nix/store/qf1sv0zy8prp39v74wjqgcrb41570mmi-iscan-gt-f720-bundle-2.30.4.x64.rpm.tar.gz.drv'...
building '/nix/store/gfx46padnajbdg6b21az60wiic3c87c3-iscan-gt-s650-bundle-2.30.4.x64.rpm.tar.gz.drv'...
building '/nix/store/3cv0902px1zhgbpqff24zgs4jw3vpdgj-iscan-gt-s80-bundle-2.30.4.x64.rpm.tar.gz.drv'...
building '/nix/store/4rvdspgkw2kznzqyiq5y1al2bwbwj2p9-iscan-gt-x770-bundle-2.30.4.x64.rpm.tar.gz.drv'...
building '/nix/store/sgfqy0l7zxaqn59zii5v5ghb79p4m1ns-iscan-perfection-v330-bundle-2.30.4.x64.rpm.tar.gz.drv'...
unpacking sources
unpacking source archive /nix/store/qrns8xnxyp7n0zlk9shkbplj7f996d6c-iscan-data_1.39.1-2.tar.gz
source root is iscan-data-1.39.1
setting SOURCE_DATE_EPOCH to timestamp 1566342420 of file iscan-data-1.39.1/iscan-data.spec
patching sources
configuring
configure flags: --prefix=/nix/store/ffla9lgmmnfk9kqn659651g47xy3fbak-iscan-data-1.39.1-2
checking for a BSD-compatible install... /nix/store/xnks3zfwvvqgdb2lplg3s3v41k5nnwaz-coreutils-8.31/bin/install -c

trying https://download2.ebz.epson.net/iscan/plugin/gt-s650/rpm/x64/iscan-gt-s650-bundle-2.30.4.x64.rpm.tar.gz

trying https://download2.ebz.epson.net/iscan/general/rpm/x64/iscan-bundle-2.30.4.x64.rpm.tar.gz

trying https://download2.ebz.epson.net/iscan/plugin/gt-s80/rpm/x64/iscan-gt-s80-bundle-2.30.4.x64.rpm.tar.gz

trying https://download2.ebz.epson.net/iscan/plugin/gt-f720/rpm/x64/iscan-gt-f720-bundle-2.30.4.x64.rpm.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

trying https://download2.ebz.epson.net/iscan/plugin/gt-x770/rpm/x64/iscan-gt-x770-bundle-2.30.4.x64.rpm.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

trying https://download2.ebz.epson.net/iscan/plugin/perfection-v330/rpm/x64/iscan-perfection-v330-bundle-2.30.4.x64.rpm.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  615k  100  615k    0     0  2888k      0 --:--:-- --:--:-- --:--:-- 2874k
100  490k  100  490k    0     0  2200k      0 --:--:-- --:--:-- --:--:-- 2191k
hash mismatch in fixed-output derivation '/nix/store/8rbxcdbw2hp014hn8wf7hl9vlqnpiixa-iscan-bundle-2.30.4.x64.rpm.tar.gz':
  wanted: sha256:1l0y4dy88y91jdq66pxrxqmiwsxwy0rd7x4bh0cw08r4iyhjqprz
  got:    sha256:0jssigsgkxb9i7qa7db291a1gbvwl795i4ahvb7bnqp33czkj85k
hash mismatch in fixed-output derivation '/nix/store/m765wih2c15kargfcnf5l0dxsvv0wk7k-iscan-gt-x770-bundle-2.30.4.x64.rpm.tar.gz':
  wanted: sha256:1cz4z3wz216s77z185m665jcgdslil5gn4dsi118nv1fm17z3jik
  got:    sha256:1chxdm6smv2d14pn2jl9xyd0vr42diy7vpskd3b9a61gf5h3gj03
cannot build derivation '/nix/store/02lw4k8mpi9wrjvcksjbqn6q98b5ab0s-iscan-gt-x770-bundle-2.30.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/n23vabjdm2nhkmh5y4i5w7qmzwk7z9ij-iscan-nt-bundle-2.30.4.drv': 1 dependencies couldn't be built
cannot build derivation '/nix/store/k3j68jk2ssyg82q44kp5gmnxmk0z4136-iscan-2.30.4-2.drv': 1 dependencies couldn't be built
error: build of '/nix/store/k3j68jk2ssyg82q44kp5gmnxmk0z4136-iscan-2.30.4-2.drv' failed
```

**Notify maintainers**

- @symphorien
- @dominikh

**Metadata**

 - system: `"x86_64-linux"`
 - host os: `Linux 5.4.52, NixOS, 20.03.2621.190e79d0f38 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.6`
 - channels(kea): `""`
 - channels(root): `"home-manager-20.03, nixos-20.03.2621.190e79d0f38, nixos-19.09-19.09.2520.289466dd6a1, nixos-unstable-20.09pre235279.5717d9d2f7c"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
- pkgs.epkowa
# a list of nixos modules affected by the problem
module:
```

**Comment by symphorien** [comment]
I wonder if it's a bug in fetchurl that it does not try all the urls until it finds one with matching hash.

**Comment by symphorien** [comment]
hum apparently wayback machine shapshots don't have the same hash as the original webpage even if I retrigger a snapshot.

**Comment by pacien** [comment]
This has been fixed on master (309b881) and release-20.03 (13fe7cf).
